### ISSUE TITLE
Google: Searching "regex exclude" redirects to homepage with an error

### ISSUE THREAD

**Issue by dsmith2001** [searxng/searxng#2943]
**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**
2023.10.13+b013cbb4c

**How did you install SearXNG?**
searxng-docker

**What happened?**
I searched `regex exclude` and I got redirected to the homepage with a "search error" popping up.

**How To Reproduce**
- Enable the Google engine or **!go regex exclude**. 
- Mobile UI is enabled (not sure if that changes anything but I'll mention it anyways)

**Expected behavior**
- No redirection to homepage with an error. 
- Results on how to exclude keywords from regex.

**Screenshots & Logs**
![image](https://github.com/searxng/searxng/assets/129474298/11b0ece5-1ef8-4193-8579-faa3dc3a04cf)

**Comment by return42** [comment]
I can reproduce the issue (only) when I use the en language `:en !go regex exclude`

```
Traceback (most recent call last):
  File "SearXNG/searx/webapp.py", line 660, in search
    result_container = search.search()
  File "SearXNG/searx/search/__init__.py", line 210, in search
    self.result_container.close()
  File "SearXNG/searx/results.py", line 358, in close
    result['content'] = utils.html_to_text(result['content']).strip()
  File "SearXNG/searx/utils.py", line 161, in html_to_text
    s.feed(html_str)
  File "/usr/lib/python3.8/html/parser.py", line 111, in feed
    self.goahead(0)
  File "/usr/lib/python3.8/html/parser.py", line 179, in goahead
    k = self.parse_html_declaration(i)
  File "/usr/lib/python3.8/html/parser.py", line 264, in parse_html_declaration
    return self.parse_marked_section(i)
  File "/usr/lib/python3.8/_markupbase.py", line 159, in parse_marked_section
    self.error('unknown status keyword %r in marked section' % rawdata[i+3:j])
  File "/usr/lib/python3.8/_markupbase.py", line 33, in error
    raise NotImplementedError(
NotImplementedError: subclasses of ParserBase must override error()
```

https://github.com/searxng/searxng/blob/b013cbb4cc47f224f3785019ab4e56eeb264944f/searx/utils.py#L157-L161

**Comment by return42** [comment]
Its an issue in cPython that is fixed in python 3.10:

- https://github.com/python/cpython/issues/76025
- https://github.com/python/cpython/pull/8562
- https://bugs.python.org/issue31844

I will prepare a fix to handle this issue in `version <py3.10` ..
